transcom/mymove
# Import Jan 2020 discount rates

## Description

This PR contains a migration to import the discount rates for the first performance period of 2020 (Jan 1-May 14).  A local migration with scrubbed data is included in the PR, and a secure migration for deployed environments has been uploaded to S3.

## Reviewer Notes

We currently only have a single data file with the discount rates for each TDL for this period.  We do not know which TSP had those rates, nor the BVS that was associated with it.  That shouldn't be an issue for PPM calculations, but our current database schema requires a TSPP record to have a TSP and BVS.   In case we're not able to get that data before we need to deploy, I have set the BVS to 100 and the TSP to the Truss test TSP (TRS1).  I don't believe that will cause any problems in calculations since that was really only useful for HHGs, but please consider if this is an issue.  If we do get this data later and want to adjust the BVS/TSP, we should be able to write another migration to update the records inserted by this PR.  See these two Slack threads for more details:  [thread 1](https://ustcdp3.slack.com/archives/CQ08Q9FPY/p1576695533003900), [thread 2](https://ustcdp3.slack.com/archives/CP6F568DC/p1577461707007500).

One note on the Truss TSP -- it uses a different UUID locally vs. the deployed environments, so you will see a variation if you compare the two migrations.

I plan to deploy this to experimental after getting an initial thumbs up on the PR to make sure it works OK there before it lands in staging/prod.

*UPDATE*: After further discussion with the team and product, I've changed this PR so it has a basic migration to create a special TSP (`PPMX`) to reference rather than using the `TRS1` test TSP mentioned above.

## Setup

To test locally (your `dev_db` database):
- `make db_dev_reset db_dev_migrate`

To test deployed migrations (your `deployed_migrations` database):
- `make run_prod_migrations` (or `run_staging_migrations` or `run_experimental_migrations`)

You should find 847 `transportation_service_provider_performances` records with a `performance_period_start` of `2020-01-01` in the databases above.

Please also spot check the discount rates in those records vs. those found in the data file linked below.

## Code Review Verification Steps

* Any new migrations/schema changes:

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-941) for this change
* [Directory with data files](https://drive.google.com/drive/u/0/folders/1gi42HBMu7jPKDUhSD-OWW9xczI770k7G) -- see ones with `1Jan20` in the name
